### ISSUE TITLE
The AI sometimes starves

### ISSUE THREAD

**Issue by Kistrel** [Kromster80/kam_remake#73]
(version r6720)

I've noticed that the AI, if left on its own and with no peacetime, tends to make little food for the citizens and troops it recruits, and sometimes ends up starving.

It probably isn't really an issue (better realism and better town management when its troops aren't left alone by the player?), just thought it would be something worth reporting anyway.

**Comment by Kromster80** [comment]
Please attach a savegame/replay that has this issue reproducing.

**Comment by Kistrel** [comment]
As I wrote, it isn't really an issue... in most included maps it won't happen noticeably, if at all.
I noticed it because it happens in a map I modified to make the AI do its best (I took the map Mega Land, added a script that keeps replenishing the land under the AI's gold mines -since it cannot buy gold from the Market-, and set the AI to train as many soldiers as it can, both Wooden and Iron); with those settings, its food production doesn't seem to keep up with its army growth.

To put it in other words, I suppose that the AI doesn't scale its food production to either its recruiting settings or its auto-defence army count.
Again, it isn't really an issue... just something I thought would be worth reporting.

Should I upload a replay and the modified map anyway?

**Comment by Kromster80** [comment]
I would assume AI should not feed troops that are planned to be used in attacks. Does the AI attack correspondingly often?

**Comment by Kistrel** [comment]
No, I don't think so... keep in mind though that this was observed on a series of 5 or 6 "2 humans vs 5 AI" games with a friend, so the number of observations isn't very high.
Anyway, the observed behaviour is that when it has food, it attacks often and hard; when it is starving, it never attacks.
This relationship could be reversed, or could both be related to other factors; for example, in one game where it didn't have food problems at all and attacked often, my friend had misplanned his town and ended up with no army for quite a bit.
My troops were his only line of defence: is it possible that the AI kept attacking him because it considered him to be a weak target, and the constant slaughter of its troops kept its food reserves high, which in turn prevented its town from halting production due to starvation?
(In that game the AI attacked me very little, the first time being when I ended up with no troops after first sending everyone to my ally; it's true that I had destroyed the closest town early, but one of the 4 remaining mostly attacked my friend even though I was the closest to its town).
